src/go/rpk/pkg/cli/cmd/cloud/auth/config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package auth
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Config represents the cloud configuration used to log in to the Redpanda
+// Cloud.
+type Config struct {
+	path string
+
+	ClientID     string `yaml:"client_id,omitempty"`
+	ClientSecret string `yaml:"client_secret,omitempty"`
+	AuthToken    string `yaml:"auth_token,omitempty"`
+}
+
+// defaultCfgPath returns the default path where the cloud configuration will
+// live, normally: '$HOME/.config/rpk/cloud.yaml'.
+func defaultCfgPath() (string, error) {
+	dir, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("unable to get home dir: %w", err)
+	}
+	return filepath.Join(dir, ".config", "rpk", "cloud.yaml"), nil
+}
+
+// LoadConfig loads the cloud configuration in the default path.
+func LoadConfig() (*Config, error) {
+	path, err := defaultCfgPath()
+	if err != nil {
+		return nil, err
+	}
+
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load config file at path %s: %w", path, err)
+	}
+
+	var cfg Config
+	if err := yaml.Unmarshal(contents, &cfg); err != nil {
+		return nil, fmt.Errorf("unable to decode config file at path %s: %w", path, err)
+	}
+
+	cfg.path = path
+
+	return &cfg, nil
+}
+
+// Save writes the config to disk at its previously specified path or at the
+// default path if the file does not exist yet.
+func (c *Config) Save() error {
+	contents, err := yaml.Marshal(c)
+	if err != nil {
+		return fmt.Errorf("unable to encode config file: %w", err)
+	}
+
+	if c.path == "" {
+		path, err := defaultCfgPath()
+		if err != nil {
+			return err
+		}
+		c.path = path
+	}
+	dir := filepath.Dir(c.path)
+	if err := os.MkdirAll(dir, 0o755); err != nil {
```

Comment:
Yes, however, I think we should check if we are running with `sudo` at the beginning of the LoadFlow, otherwise, the user will have to enter their credentials and exchange the token to then realize (in cfg.Save) that they shouldn't run this with sudo.